src/index.js

```diff
@@ -25,10 +25,84 @@ import cors from 'cors';
 import ip from 'ip';
 import ProxyAgent from 'proxy-agent';
 
+const moduleNames = [];
+const cacheDirectory = process.env.HOMEPATH || process.env.HOME;
+const fileName = '.one-app-module-cache';
+const filePath = path.join(cacheDirectory, fileName);
+
+// show cache size and how to delete info on start
+const showCacheInfo = () => {
+  fs.stat(filePath, (error, stats) => {
+    if (error) {
+      console.log('There was error checking file stat', error);
+    }
+    const fileSizeOnMB = stats.size / (1024 * 1024); // bytes to mb
+    console.log(`File size of ${fileName}: ${fileSizeOnMB.toFixed(2)} MB`);
+    console.log(`To delete cache, please run \nrm ${filePath}`);
+  });
+};
+
+// gets cached module from .one-app-module-cache
+const getCachedModules = () => {
+  let hasCachedFile = false;
+  try {
+    fs.accessSync(filePath, fs.constants.F_OK);
+    hasCachedFile = true;
+  } catch (error) {
+    console.log(`Creating ${fileName} on ${cacheDirectory}`);
+    try {
+      fs.writeFileSync(filePath, JSON.stringify('{}'));
+      console.log(`${fileName} created successfully on ${cacheDirectory}`);
+    } catch (err) {
+      console.error(`Error creating ${fileName} on ${cacheDirectory}, \n${err}`);
+    }
+  }
+  if (hasCachedFile) {
+    try {
+      showCacheInfo();
+      const cachedContent = fs.readFileSync(filePath, 'utf8');
+      return JSON.parse(cachedContent);
+    } catch (err) {
+      console.error('Could not parse JSON content', err);
+    }
+  }
+  return {};
+};
+
+const cachedModules = getCachedModules();
+
+let timerId = null;
+
+const setOnCache = (content, delay = 500) => {
+  // added debounce
+  clearTimeout(timerId);
+  timerId = setTimeout(() => {
+    fs.writeFile(filePath, JSON.stringify(content, null, 2), (error) => {
+      if (error) {
+        console.log(`There was an error updating content \n ${error}`);
+      }
+    });
+    timerId = null;
```

Comment:
This does not add anything as javascript discards `timerId` once it executes `setTimeout`. But, it helps with code clarity.